unit_tests/test_aggregator.py

```diff
@@ -52,31 +44,21 @@ def mock_start_websocket_funcs(mocker):
 
 @pytest.fixture(autouse=True)
 def mock_execute_batch_command_methods(mocker):
-    """
-    Fixture mocks methods and external functions and other methods used in
-    execute_batch_command.
-    """
+    """Mock methods and external functions and other methods used in execute_batch_command."""
     mocker.patch("gsy_e_sdk.aggregator.ClientCommandBuffer.buffer_length",
                  new_callable=PropertyMock, return_value=3)
     mocker.patch("gsy_e_sdk.aggregator.ClientCommandBuffer.execute_batch",
                  return_value=test_batch_command_dict)
     mocker.patch("gsy_e_sdk.aggregator.Aggregator._post_request",
                  return_value=["some_transaction_id", True])
-    mocker.patch("gsy_e_sdk.aggregator.AggregatorWebsocketMessageReceiver"
-                 + ".wait_for_command_response",
-                 return_value=test_response)
     mocker.patch("gsy_e_sdk.aggregator.Aggregator."
                  + "_all_uuids_in_selected_device_uuid_list",
                  return_value=True)
-    mocker.patch("gsy_e_sdk.aggregator.log_bid_offer_confirmation")
```

Comment:
This seems to be only be used once, so I patched it in this specific use case, same for the `log_deleted_bid_offer_confirmation`

Reply:
Is the first modification after the function-docstring the one you meant earlier about reverting the patch in this fixture?

Reply:
This is a similar case as the one in the first comment. I decided to patch this way, because in other tests cases it might be used when calling `execute_batch_commands()`. When I want to test explicitly if these functions were called is when I invoke the context manager.

Reply:
I understand that the `auto_use=True` clause takes this fixture to every test case. Originally I wanted to avoid this by calling it only on the test_execute_batch_commands__*() test cases, however, as these fixtures do not return anything I had to deal with the warning about not using a passed variable. 

In order to avoid passing the fixture to the test case of interest (and therefore avoiding the warning) I set `auto_use=True`.

I would really appreciate any suggestions regarding this I mentioned.

Reply:
You can try the `@pytest.mark.usefixtures('my_fixture')` decorator, and in any case, and please always five priority to functional code quality over pylint's warnings, pylint mainly pushes us towards better code but sometimes it could be non supporting to other libraries (this could be clear in tests)

Reply:
I'll have a look to this you mention.